contents/projectManagement/testDrivenDevelopment.md

```diff
@@ -0,0 +1,232 @@
+
+1. **Strict emphasis on communication of requirements and specifications**
+
+    Product/Feature requirements are crucial to teams that employ a TDD-style approach to development. As mentioned earlier, the very first step of the development cycle is to write acceptance tests. If the requirements are not communicated properly, teams will end up constantly having to change their test suites. In extreme scenarios, a re-write may be necessary.
+
+    As a result, a lot of man hours can go into making sure the requirements are in order -  consequently, this could result in a slow start to development.
+
+2. **Tough to incorporate TDD into <trigger trigger="click" for="modal:index-legacy">legacy projects [^6]</trigger>**
+
+    Legacy projects are hard to maintain even without having to incorporate TDD into them. As these projects have a history of design and implementation choices made based on the requirements and/or the resources at the time, investing man hours in revamping or "modernizing" them is not always deemed as top priority. 
+
+    <modal title="Legacy Code" id="modal:index-legacy">
+    Legacy code refers to an application system source code type that is no longer supported. Legacy code can also refer to unsupported operating systems, hardware and formats. In most cases, legacy code is converted to a modern software language and platform. However, to retain familiar user functionality, legacy code is sometimes carried into new environments. 
+    </modal>
+
+    When you bring TDD into the picture, things get more complex as failing tests are written at first with the assumption that production code will be written to ensure they pass. 
+    
+    Writing production code in legacy systems with the goal of passing a specific test is harder than it seems. For instance, there may be dependencies between components that need to be "broken" for them to be testable, which could result in  several workarounds. This could potentially add on to the complexity of the project, making it counter-productive as the entire point of the refactoring step is to make the code more readable[^4].
+
+3. **High Maintenance**
+    
+    Maintaining test-suites becomes a part of the overall overhead of the project. Since TDD-style approaches rely heavily on tests, there is a need for the tests to be in proper shape in order to prevent a false sense of correctness. 
+    
+    TDD also has a steep learning curve. The entire process takes time to get accustomed to. Refactoring of code by using good design patterns comes with time, experience, and a lot of practice.
+
+4. **Great Importance given to Unit Tests**    
+
+    Unit tests are generally much more in number than other sort of tests like integration, system, or interface tests. Developers run the risk of giving less importance to these classes of tests if there are a high number of passing unit tests[^3].
+
+## Advanced Test-Driven Development
```

Comment:
Although I said it is OK to keep this section, now that I'm reading this again, kind of hard to justify having this here as this is supposed to be an 'intro to TDD' rather than 'summary of TDD'. Can't see how someone who doesn't even know TDD will benefit reading this part. Remove?

Reply:
I guess it's more of an FYI. It's not really information that the user _needs to know_. 
Would it be better to add this as an optional reading suggestion for those interested?

Reply:
Yes, if at all, it an be a link to another article.